src/main/java/org/kitteh/irc/client/library/feature/twitch/TwitchSupport.java

```diff
@@ -0,0 +1,162 @@
+import org.kitteh.irc.client.library.feature.twitch.messagetag.MsgParamMonths;
+import org.kitteh.irc.client.library.feature.twitch.messagetag.MsgParamSubPlan;
+import org.kitteh.irc.client.library.feature.twitch.messagetag.MsgParamSubPlanName;
+import org.kitteh.irc.client.library.feature.twitch.messagetag.R9k;
+import org.kitteh.irc.client.library.feature.twitch.messagetag.RoomId;
+import org.kitteh.irc.client.library.feature.twitch.messagetag.Slow;
+import org.kitteh.irc.client.library.feature.twitch.messagetag.SubsOnly;
+import org.kitteh.irc.client.library.feature.twitch.messagetag.Subscriber;
+import org.kitteh.irc.client.library.feature.twitch.messagetag.SystemMsg;
+import org.kitteh.irc.client.library.feature.twitch.messagetag.Turbo;
+import org.kitteh.irc.client.library.feature.twitch.messagetag.User;
+import org.kitteh.irc.client.library.feature.twitch.messagetag.UserId;
+import org.kitteh.irc.client.library.feature.twitch.messagetag.UserType;
+
+/**
+ * A class for introducing Twitch support to a KICL client.
+ */
+public final class TwitchSupport {
+    /**
+     * Capability to receive commands.
+     */
+    public static final String CAPABILITY_COMMANDS = "twitch.tv/commands";
+
+    /**
+     * Capability to receive JOIN, MODE, NAMES, and PART.
+     */
+    public static final String CAPABILITY_MEMBERSHIP = "twitch.tv/membership";
+
+    /**
+     * Capability to receive tags.
+     */
+    public static final String CAPABILITY_TAGS = "twitch.tv/tags";
+
+    /**
+     * Adds support for Twitch to the client. Registers all the Twitch tags,
+     * switches out an event listener for connection information (Twitch does
+     * it differently), disables querying channel info (missing on Twitch),
+     * and replaces the message sending queue supplier with a Twitch-targeted
+     * queue for regular users.
+     *
+     * @param builder client builder
+     * @return built client, not yet connected
+     * @see TwitchListener
+     * @see TwitchDelaySender
+     */
+    public static Client buildWithSupport(Client.@NonNull Builder builder) {
+        return buildWithSupport(builder, false);
+    }
+
+    /**
+     * Adds support for Twitch to the client. Registers all the Twitch tags,
+     * switches out an event listener for connection information (Twitch does
+     * it differently), disables querying channel info (missing on Twitch),
+     * and sets the message sending queue supplier to be a Twitch-targeted
+     * queue for regular users or mod/op depending on the second parameter.
+     * This method must be called prior to connecting.
+     *
+     * @param builder client builder
+     * @param alwaysModOrOp if the client will be a mod/op in EVERY channel
+     * @return built client, not yet connected
+     * @see TwitchListener
+     * @see TwitchDelaySender
+     */
+    public static Client buildWithSupport(Client.@NonNull Builder builder, boolean alwaysModOrOp) {
```

Comment:
I think this should be a method that just modifies a client. `void addSupport(Client, ..)`.

I don't like the idea of giving up control of building like this.

Reply:
`boolean isConnectionAlive()` added and checked.